files/src/main/scala/io/iteratee/files/FileModule.scala

```diff
@@ -18,37 +18,45 @@ import scala.collection.JavaConverters._
 import scala.util.{ Left, Right }
 
 trait FileModule[F[_]] { this: Module[F] { type M[f[_]] <: MonadError[f, Throwable] } =>
-  protected def captureEffect[A](a: => A): F[A]
+  protected def captureEffect[A](a: => A): F[A] = F.catchNonFatal(a)
```

Comment:
@johnynek What do you think about providing this operation via a new type class, so we could follow the pattern in iteratee-core, where all of the operations are available on a `FileEnumerators` object, and `FileModule` is just a syntactic convenience? The implementations would be more or less the same, and that approach would be likely to involve fewer API changes if Cats introduces a more general solution to the problem.